dotcom-rendering/src/components/ProductElement.tsx

```diff
@@ -0,0 +1,153 @@
+import { css } from '@emotion/react';
+import { from } from '@guardian/source/foundations';
+import type { ReactNode } from 'react';
+import type { ArticleFormat } from '../lib/articleFormat';
+import { parseHtml } from '../lib/domUtils';
+import type { NestedArticleElement } from '../lib/renderElement';
+import type {
+	FEElement,
+	ProductBlockElement,
+	ProductCta,
+	ProductDisplayType,
+	ProductImage,
+} from '../types/content';
+import { ProductCardInline } from './ProductCardInline';
+import { ProductCardLeftCol } from './ProductCardLeftCol';
+import { buildElementTree } from './SubheadingBlockComponent';
+
+export type Product = {
+	primaryHeadline: string;
+	secondaryHeadline: string;
+	brandName: string;
+	productName: string;
+	image: ProductImage;
+	retailer: string;
+	customAttributes: {
+		name: string;
+		value: string;
+	}[];
+	content: FEElement[];
+	displayType: ProductDisplayType;
+	productCtas: ProductCta[];
+};
+
+const contentContainer = css`
+	position: relative;
+`;
+
+const LeftColProductCardContainer = ({ children }: { children: ReactNode }) => (
+	<div
+		css={css`
+			display: none;
+			${from.wide} {
+				position: absolute;
+				left: -240px;
+				height: 100%;
+				display: block;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+export const ProductElement = ({
+	product,
+	ArticleElementComponent,
+	format,
+	shouldShowLeftColCard,
+}: {
+	product: ProductBlockElement;
+	ArticleElementComponent: NestedArticleElement;
+	format: ArticleFormat;
+	shouldShowLeftColCard: boolean;
+}) => {
+	const showContent =
+		product.displayType === 'InlineOnly' ||
+		product.displayType === 'InlineWithProductCard';
+	const showProductCard =
+		product.displayType === 'ProductCardOnly' ||
+		product.displayType === 'InlineWithProductCard';
+	return (
+		<div>
+			{showContent && (
+				<Content
+					product={product}
+					format={format}
+					ArticleElementComponent={ArticleElementComponent}
+					shouldShowLeftColCard={shouldShowLeftColCard}
+				/>
+			)}
+			{showProductCard && (
+				<ProductCardInline
+					format={format}
+					brandName={product.brandName}
+					productName={product.productName}
+					image={product.image}
+					lowestPrice={product.lowestPrice}
+					productCtas={product.productCtas}
+					customAttributes={product.customAttributes}
+					isCardOnly={product.displayType === 'ProductCardOnly'}
+					shouldShowLeftColCard={shouldShowLeftColCard}
+				/>
+			)}
+		</div>
+	);
+};
+
+const Content = ({
+	product,
+	ArticleElementComponent,
+	format,
+	shouldShowLeftColCard,
+}: {
+	product: ProductBlockElement;
+	ArticleElementComponent: NestedArticleElement;
+	format: ArticleFormat;
+	shouldShowLeftColCard: boolean;
+}) => {
+	const showLeftCol =
+		product.displayType === 'InlineWithProductCard' &&
+		shouldShowLeftColCard;
+	const subheadingHtml = parseHtml(
+		`<h2 id="${product.h2Id ?? product.elementId}">${
+			product.primaryHeading ? `${product.primaryHeading}</br>` : ''
```

Comment:
A minor detail, but `<br>` is a self-closing tag so should be `<br />` rather than `</br>`